packages/common/tests/hardforks.spec.ts

```diff
@@ -63,6 +63,86 @@ tape('[Common]: Hardfork logic', function (t: tape.Test) {
     st.end()
   })
 
+  t.test('test post merge hardforks using Sepolia', function (st: tape.Test) {
+    const c = new Common({ chain: Chain.Sepolia })
+    let msg = 'should get HF correctly'
+
+    st.equal(c.getHardforkByBlockNumber(0), Hardfork.London, msg)
+    //using Hardfork.London even though Merge is expected after terminal block 1450408
+    st.equal(c.getHardforkByBlockNumber(1450409), Hardfork.London, msg)
+    st.equal(c.getHardforkByBlockNumber(1450409, BigInt('17000000000000000')), Hardfork.Merge, msg)
```

Comment:
I think this result is counterintuitive.  The function is called `getHardforkByBlockNumber` and yet when we set TTD and a block number prior to the actual merge block, it's telling me the hardfork is the Merge hardfork.  If we're checking the hardfork and specifying a block number prior to the actual block where the merge happens, we shouldn't report a hardfork that is happening later.  I know there's screwiness around Merge and MergeForkTransitionId but this doesn't make a ton of sense to me.

Reply:
I still think this result is counterintuitive.  If you provide a block number and the merge hasn't happened yet, regardless of what you pass in for TD, this method should return the previous hardfork.  Block number should trump hardfork in terms of which fork is returned.

Reply:
merge HAS happened if we provide td > ttd (L73), and hasn't happened if we don't provide td L72